src/wire/ipv4.rs

```diff
@@ -7,6 +7,11 @@ use super::ip::{checksum, pretty_print_ip_payload};
 
 pub use super::IpProtocol as Protocol;
 
+/// Minimum MTU required of all links supporting IPv4. See [RFC 791 § 3.1]
+///
+/// [RFC 791 § 3.1]: https://tools.ietf.org/html/rfc791#section-3.1
+pub const MIN_MTU: usize = 576;
```

Comment:
No, this is actually not enough, 576 is the minimum datagram size. If you have e.g. an Ethernet header in front, then MTU of 576 is not sufficient.

Reply:
Right... my bad

Reply:
It is the minimum MTU for IPv6 https://tools.ietf.org/html/rfc8200#section-5. I think that is where I got confused. I'll have to think about how to best word this, since it is the minimum datagram size for IPv4 and the minimum MTU for IPv6. Maybe we end up with `IPV4_MIN_DATAGRAM_SIZE` and `IPV6_MIN_MTU`?

Reply:
Still not quite, it's very confusing. Take a look at https://en.wikipedia.org/wiki/Maximum_transmission_unit#MTUs_for_common_media. The minimum MTU on IPv4 is actually 68, but since we never fragment ourselves, the *effective* minimum MTU supported by smoltcp on IPv4 is 576. This needs to be documented, and the constant can remain the same.